pkg/controller/account/account_controller.go

```diff
@@ -261,6 +261,42 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 	}
 
+	var roleToAssume string
+	var currentAccInstanceID string
+	var iamUserUHC = iamUserNameUHC
+	var iamUserSRE = iamUserNameSRE
+
+	if currentAcctInstance.IsBYOC() {
+		// Use the same ID applied to the account name for IAM usernames
+		currentAccInstanceID = currentAcctInstance.Labels[awsv1alpha1.IAMUserIDLabel]
+		iamUserUHC = fmt.Sprintf("%s-%s", iamUserNameUHC, currentAccInstanceID)
+		iamUserSRE = fmt.Sprintf("%s-%s", iamUserNameSRE, currentAccInstanceID)
+		roleToAssume = fmt.Sprintf("%s-%s", byocRole, currentAccInstanceID)
+	} else {
+		roleToAssume = awsv1alpha1.AccountOperatorIAMRole
+	}
+
+	awsAssumedRoleClient, creds, err := r.assumeRole(reqLogger, currentAcctInstance, awsSetupClient, roleToAssume, ccsRoleID)
+	if err != nil {
+		// assumeRole logs
+		return reconcile.Result{}, err
+	}
+
+	// Make sure every cluster has the AWSManaged role
+	SREAccessARN, err := r.GetSREAccessARN(reqLogger)
+	if err != nil {
+		reqLogger.Error(err, "There was an error retrieving the SRE access ARN")
+		return reconcile.Result{}, err
+	}
+
+	tags := awsclient.AWSTags.BuildTags(currentAcctInstance).GetIAMTags()
+	roleID, err := CreateAWSManagedRole(reqLogger, awsSetupClient, awsAssumedRoleClient, awsv1alpha1.AWSManagedRoleArn, tags, SREAccessARN)
+	if err != nil {
+		reqLogger.Error(err, "Failed to sync AWSManagedRole")
+		return reconcile.Result{}, err
+	}
+	reqLogger.Info("AWSManagedRole created", "RoleID", roleID, "account", currentAcctInstance.Name)
+
```

Comment:
I think this block needs to get moved back down to where it was, as we explicitly do _not_ want to create roles in STS-based accounts.

Reply:
The reason I moved it out is because I thought that since it was inside `currentAcctInstance.ReadyForInitialization()` it would only create the role when accounts are initialized and not create roles in old accounts. I could be wrong? Would an alternative just be checking `currentAcctInstance.Spec.ManualSTSMode`?

Reply:
I updated this to not create for STS accounts